on-this-day-api/controllers/fact-controller.js

```diff
@@ -0,0 +1,33 @@
+const factModel = require('../models/fact-model');
+
+async function getFact (ctx) {
+  const date = ctx.url.slice(-5);
+  const obj = { date: date };
+  if (!ctx || date.includes('/') || date === '02-30' || date === '02-31' || date === '04-31' || date === '06-31' || date === '09-31' || date === '11-31') {
+    ctx.status = 404;
+    return ctx.response.body = "Incorrect url. Check the date."
+  };
+  try {
```

Comment:
if you were using express you could create a try catch middleware to avoid writting try catch blocks on your handlers:
```
  Promise
    .resolve(fn(req, res, next))
    .catch(next)
app.get('/hello', asyncHandler( (req, res, next) => {
  // Some code here. Any error will be catch and pass to expressjs
}) );
but I believe koa is allready doing this for you, so if you dont want to do something specific on the catch you can remove the try catch

Reply:
https://github.com/koajs/koa/wiki/Error-Handling